homeassistant/components/sensor/syncthru.py

```diff
@@ -60,32 +61,37 @@
 })
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
     """Set up the SyncThru component."""
-    from pysyncthru import SyncThru, test_syncthru
+    from pysyncthru import SyncThru
 
     if discovery_info is not None:
+        _LOGGER.info("Discovered a new Samsung Printer at %s",
+                     discovery_info.get(CONF_HOST))
         host = discovery_info.get(CONF_HOST)
         name = discovery_info.get(CONF_NAME, DEFAULT_NAME)
-        _LOGGER.debug("Discovered a new Samsung Printer: %s", discovery_info)
-        # Test if the discovered device actually is a syncthru printer
-        if not test_syncthru(host):
-            _LOGGER.error("No SyncThru Printer found at %s", host)
-            return
+        # Main device, always added
         monitored = DEFAULT_MONITORED_CONDITIONS
     else:
         host = config.get(CONF_RESOURCE)
         name = config.get(CONF_NAME)
         monitored = config.get(CONF_MONITORED_CONDITIONS)
 
-    # Main device, always added
+    session = aiohttp_client.async_get_clientsession(hass)
+
+    printer = SyncThru(host, session)
+    # Test if the discovered device actually is a syncthru printer
     try:
-        printer = SyncThru(host)
-    except TypeError:
-        # if an exception is thrown, printer cannot be set up
+        # No error is thrown when the device is off
+        # (only after user added it manually)
+        # therefore additional catches are inside the Sensor below
+        await printer.update()
+    except ValueError:
+        # if an exception is thrown, printer does not support syncthru
+        _LOGGER.info("Samsung printer at %s does not support SyncThru", host)
```

Comment:
Shouldn't this be a warning?

Reply:
No, the printer can be discovered automatically but still may not support syncthru, then no warnings or errors should be thrown. Still the way I coded it here was incorrect, see new source code for correct handling.